nca/CoreDS/ConnectivityProperties.py

```diff
@@ -0,0 +1,603 @@
+                      f'of the pod {peer}. Ignoring the pod')
+                continue
+            real_ports.add_port(real_port[0])
+        return real_ports
+
+    @staticmethod
+    def make_conn_props(conn_cube):
+        """
+        This will create connectivity properties made of the given connectivity cube.
+        This includes tcp properties, non-tcp properties, icmp data properties.
+        If possible (i.e., in original solution, when dst_peers are supported), the named ports will be resolved.
+
+        In the optimized solution, the resulting ConnectivityProperties should not contain named ports:
+            they are substituted with corresponding port numbers, per peer
+        In the original solution, the resulting ConnectivityProperties may contain named ports;
+            they cannot yet be resolved, since dst peers are not provided at this stage the original solution;
+            they will be resolved by convert_named_ports call during query runs.
+
+        :param ConnectivityCube conn_cube: the input connectivity cube including all dimension values,
+            whereas missing dimensions are represented by their default values (representing all possible values).
+        """
+
+        src_ports = conn_cube.get_dim("src_ports")
+        dst_ports = conn_cube.get_dim("dst_ports")
+        dst_peers = conn_cube.get_dim("dst_peers")
+        assert not src_ports.named_ports and not src_ports.excluded_named_ports
+        if (not dst_ports.named_ports and not dst_ports.excluded_named_ports) or not dst_peers:
+            # Should not resolve named ports
+            return ConnectivityProperties(conn_cube)
+
+        # Initialize conn_properties
+        if dst_ports.port_set:
+            dst_ports_no_named_ports = dst_ports.copy()
+            dst_ports_no_named_ports.named_ports = set()
+            dst_ports_no_named_ports.excluded_named_ports = set()
+            conn_cube.set_dim("dst_ports", dst_ports_no_named_ports)
+            conn_properties = ConnectivityProperties(conn_cube)
+        else:
+            conn_properties = ConnectivityProperties.make_empty_props()
+
+        # Resolving dst named ports
+        protocols = conn_cube.get_dim("protocols")
+        assert dst_peers
+        for peer in dst_peers:
+            real_ports = ConnectivityProperties.resolve_named_ports(dst_ports.named_ports, peer, protocols)
+            if real_ports:
+                conn_cube.set_dims({"dst_ports": real_ports, "dst_peers": PeerSet({peer})})
+                conn_properties |= ConnectivityProperties(conn_cube)
+            excluded_real_ports = ConnectivityProperties.resolve_named_ports(dst_ports.excluded_named_ports, peer, protocols)
+            if excluded_real_ports:
+                conn_cube.set_dims({"dst_ports": excluded_real_ports, "dst_peers": PeerSet({peer})})
+                conn_properties -= ConnectivityProperties(conn_cube)
+        return conn_properties
+
```

Comment:
can add unit tests for this named ports resolution during construction?
currently we have unit tests at `testConnectivityPropertiesNamedPorts.py` which only tests the resolving of the original solution.